Upstream-Status: Pending
---
diff --git a/onnxruntime/core/mlas/lib/platform.cpp b/onnxruntime/core/mlas/lib/platform.cpp
index 23d29fd02f..ed48c250dc 100644
--- a/onnxruntime/core/mlas/lib/platform.cpp
+++ b/onnxruntime/core/mlas/lib/platform.cpp
@@ -569,11 +569,11 @@ Return Value:
     //
     // Check if the processor supports ASIMD I8MM instructions.
     //
-    if (MLAS_CPUIDINFO::GetCPUIDInfo().HasArmNeon_I8MM()) {
-        this->GemmU8U8Dispatch = &MlasGemmU8X8DispatchUmmla;
-        this->GemmU8S8Dispatch = &MlasGemmU8X8DispatchUmmla;
-        this->GemmS8S8Dispatch = &MlasGemmS8S8DispatchSmmla;
-    }
+    // if (MLAS_CPUIDINFO::GetCPUIDInfo().HasArmNeon_I8MM()) {
+        // this->GemmU8U8Dispatch = &MlasGemmU8X8DispatchUmmla;
+        // this->GemmU8S8Dispatch = &MlasGemmU8X8DispatchUmmla;
+        // this->GemmS8S8Dispatch = &MlasGemmS8S8DispatchSmmla;
+    // }
 #endif
 
 #if defined(MLAS_F16VEC_INTRINSICS_SUPPORTED)
