From 69d5aa765d92101ffeb018df31a255b1afc4753a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20P=C3=A9ron?= <peron.clem@gmail.com>
Date: Thu, 20 Jun 2024 11:54:03 +0200
Subject: [PATCH] arm64: force -mcpu to be valid

Upstream-Status: Submitted [https://github.com/microsoft/onnxruntime/pull/21117]
---
diff --git a/cmake/onnxruntime_mlas.cmake b/cmake/onnxruntime_mlas.cmake
index 15864a01981..5522bb9418d 100644
--- a/cmake/onnxruntime_mlas.cmake
+++ b/cmake/onnxruntime_mlas.cmake
@@ -63,6 +63,20 @@ endif()
 
 set(ONNXRUNTIME_MLAS_LIBS onnxruntime_mlas)
 
+function(remove_mcpu_flags var_name)
+  set(_original "${${var_name}}")
+  string(REGEX REPLACE "-mcpu=[^ ]*" "" _updated "${_original}")
+  if (NOT _original STREQUAL _updated)
+      message("Dropped -mcpu flags from ${var_name} updated to: ${_updated}")
+      set(${var_name} "${_updated}" PARENT_SCOPE)
+  endif()
+endfunction()
+
+# Remove -mcpu flags from ASM, C and CXX flags
+remove_mcpu_flags(CMAKE_ASM_FLAGS)
+remove_mcpu_flags(CMAKE_C_FLAGS)
+remove_mcpu_flags(CMAKE_CXX_FLAGS)
+
 #TODO: set MASM flags properly
 function(setup_mlas_source_for_windows)
 

