diff --git a/onnxruntime/core/mlas/lib/platform.cpp b/onnxruntime/core/mlas/lib/platform.cpp
index de092f7d1d..1456421729 100644
--- a/onnxruntime/core/mlas/lib/platform.cpp
+++ b/onnxruntime/core/mlas/lib/platform.cpp
@@ -526,11 +526,11 @@ Return Value:
     //
     // Check if the processor supports ASIMD I8MM instructions.
     //
-    if (MLAS_CPUIDINFO::GetCPUIDInfo().HasArmNeon_I8MM()) {
-        this->GemmU8U8Dispatch = &MlasGemmU8X8DispatchUmmla;
-        this->GemmU8S8Dispatch = &MlasGemmU8X8DispatchUmmla;
-        this->GemmS8S8Dispatch = &MlasGemmS8S8DispatchSmmla;
-    }
+    // if (MLAS_CPUIDINFO::GetCPUIDInfo().HasArmNeon_I8MM()) {
+        // this->GemmU8U8Dispatch = &MlasGemmU8X8DispatchUmmla;
+        // this->GemmU8S8Dispatch = &MlasGemmU8X8DispatchUmmla;
+        // this->GemmS8S8Dispatch = &MlasGemmS8S8DispatchSmmla;
+    // }
 #endif
 
 #endif // MLAS_TARGET_ARM64
