Upstream-Status: Pending
---
diff --git a/setup.py b/setup.py
index ced2f28e387..0ac982af0fe 100644
--- a/setup.py
+++ b/setup.py
@@ -791,10 +791,6 @@ setup(
     extras_require=extras_require,
     python_requires=">=3.10",
     keywords="onnx machine learning",
-    entry_points={
-        "console_scripts": [
-            "onnxruntime_test = onnxruntime.tools.onnxruntime_test:main",
-        ]
-    },
+    entry_points={},
     classifiers=classifiers,
 )
